### ISSUE TITLE
NiTs in API document

### ISSUE THREAD

**Issue by gorryfair** [taps-api/drafts#140]
If somone agrees with these please add, otherwise let's discuss:-).

---
- a hard to parse sentence
OLD:
similarly, as errors, just as any other Events,
NEW:
similarly. Errors, just as any other Events,
or
/similarly, because Errors, just as any other Events,/
---
- To me this seems like an odd style for a RFC.
OLD:
   We begin with the architectural design principles defined in
   [TAPS-ARCH]; from these, we derive and elaborate a set of principles
   on which the design of the interface is based.
NEW
   This section begins with the architectural 
    design principles defined in
   [TAPS-ARCH]; these are used to derive the elaborate 
   a set of principles
   on which the design of the interface is based.
- Question is /elaborate/ the best word really? Is /complete/ or /detailed/
  or /concrete/ more commonly understood?
---
Section does not contain an RFC2119 boilerplate para, yet the ID uses RFC2119 language later. (did we plan not to use RFC2119 keywords in section 5 - wand 5.3)
---
OLD:
can be cloned, in order to establish
NEW:
can be cloned to establish
- Rationale, because a reader COULD read this as implying ordering and the phrase /in order/ is not necessary to comprehend the sentence.
---
OLD:
   developers should reduce inconsistency by only using the most
NEW:
   developers ought to reduce inconsistency by only using the most
- I’d prefer to avoid using /should/ at all here to remove any chance
of misunderstanding as a requirement.
---
OLD:
   The multiple identifiers
   refer to the same endpoint.
NEW:
   The multiple identifiers
   refer to the same transport endpoint.
- actually this is a little confusing, since they are actually different network layer endpoints. I think we need to say transport explicitly.
---
TEXT:
  Note that it is possible for a set
- To me this should be a new para, it is a new thought.
---
OLD:
   Implementations of this interface should also raise
   errors in configuration as early as possible, to help ensure these
   inconsistencies are caught early in the development process.
NEW:
   Implementations of this interface should also raise any
   errors in configuration as early as possible, to help ensure these
   inconsistencies are caught early in the development process.
- add “any”, because the implementations aren’t intended just to make errors.
---
OLD:
   An implementation of this interface must provide sensible defaults
NEW:
   An implementation of this interface needs provide sensible defaults
---
OLD:
      Reliable Data Transfer: This boolean property specifies whether
      the application needs the transport protocol to ensure that data
      that data is received completely and without corruption on the other side
NEW:
      Reliable Data Transfer: This boolean property specifies whether
      the application wishes to use a transport protocol that provides
      mechanisms to help ensure that all data is received and without 
      corruption on the other side
- I’ll argue that /needs/ is actually misleading and wrong. An Internet transport can not make such promises, and we shouldn’t encourage it to think so - the app is responsible for the data. I’d be even happier with better text that avoids saying the transport is solely responsible for correctness.

**Comment by britram** [comment]
Will also cite 2119 and 8174 on ambiguity of 2119 keywords. We'll need to do a pass on these keywords before final publication.
